luci/luci-app-falter-owm/files/owm.sh

```diff
@@ -61,7 +61,9 @@ olsr4_links() {
 	json_get_var olsrInterface olsrInterface
 	json_get_var ifName ifName
 	json_select ..
-	olsr4links="$olsr4links$localIP $remoteIP $remotehost $linkQuality $ifName;"
+	if [[ "$olsrInterface" != *"wg_"* ]]; then
```

Comment:
Looks good in overall. please use single '[' as [[ is bash, but this script will run under posix-shell.

Reply:
It does not work with a simple:
```sh
        if [ "$olsrInterface" != *"wg_"* ]; then               
                olsr4links="$olsr4links$localIP $remoteIP $remotehost $linkQuality $ifName;"
        fi    
```

I don't know why.

Reply:
I assume, that `*"wg_"*` is regex-magic which has another syntax in posix-shell. Probably the `[[ ]]` does only work, because OpenWrt internally links /bin/sh to the ash, which is a lighter bash. For consistency-reason, we should keep the script with posix-features only, as the shebang also states, that its posix-compatible.

I will check for the posix-way of doing this.

Reply:
Shellcheck says this:

```
In owm.sh line 64:
        if [[ "$olsrInterface" != *"wg_"* ]]; then
           ^-- SC3010: In POSIX sh, [[ ]] is undefined.
               ^------------^ SC2154: olsrInterface is referenced but not assigned.
```
After correcting the brackets, it says this:
```
In owm.sh line 64:
        if [ "$olsrInterface" != *"wg_"* ]; then
              ^------------^ SC2154: olsrInterface is referenced but not assigned.
                                 ^-----^ SC2081: [ .. ] can't match globs. Use a case statement.
```
The wiki-article for shellcheck states a posix-compliant solution, which I would highly prefer over the bash-syntax:
https://github.com/koalaman/shellcheck/wiki/SC2081

![image](https://user-images.githubusercontent.com/43141307/148677020-7b1a9a02-c1cd-4279-9866-48165642eaed.png)

Reply:
Besides that: shouldn't that case apply for IPv6-Addresses too? Maybe the if-statement should be added some lines below in die olsrd6-function too.

Reply:
I'd rather propose to get rid of the olsrd6 function, as it'll hopefully never be used :)

Reply:
I'd like to have that matter in of removing some other function in another PR. Besides that, we still didn't have a decision on that matter in the community, so we shouldn't create facts on that for now.